src/LinuxCaptureService/LinuxCaptureServiceBase.cpp

```diff
@@ -239,8 +239,15 @@ CaptureServiceBase::CaptureInitializationResult LinuxCaptureServiceBase::DoCaptu
       producer_event_processor_.get(), &tracing_handler};
   MemoryInfoHandler memory_info_handler{producer_event_processor_.get()};
 
-  const CaptureOptions& capture_options =
-      start_stop_capture_request_waiter->WaitForStartCaptureRequest();
+  auto capture_options_or_error = start_stop_capture_request_waiter->WaitForStartCaptureRequest();
+  if (capture_options_or_error.has_error()) {
+    ORBIT_ERROR("An error occurred while waiting for start: %s",
+                capture_options_or_error.error().message());
+    client_capture_event_collector->StopAndWait();
+    TerminateCapture();
+    return CaptureInitializationResult::kFailureWhileWaitingForStart;
```

Comment:
I think there is some confusion introduced here. `CaptureInitializationResult` is supposed to be returned by `InitializeCapture`, while here we are past that.

This is visible by comparing the Linux and the Windows version. 

The Windows version has an `ORBIT_UNREACHABLE();` for `InitializeCapture` returning `kFailureWhileWaitingForStart`. Then it calls `WaitForStartCaptureRequest` and if this returns an error, it returns `StatusCode::INTERNAL`.

The Linux version does something quite different. It doesn't consider whether `InitializeCapture` can return `kFailureWhileWaitingForStart` or not (it can't, right?). Then, if `WaitForStartCaptureRequest` returns an error, it returns `kFailureWhileWaitingForStart`, which is then conditioning returning `StatusCode::INTERNAL` in the caller. 

I think we need to bring the two cases close together.

The problem is that `CaptureInitializationResult` was meant to be the result of `InitializeCapture`, but now `LinuxCaptureServiceBase::DoCapture` is adding a new case to it (`kFailureWhileWaitingForStart`) which is not related to `InitializeCapture` itself.

Reply:
+2 We also had a similar discussion on this yesterday. I thought about moving `InitializeCapture` out from `DoCapture`.

I will see whether there is a better way to make this change.

Reply:
Maybe the problem comes from the fact that `DoCapture` "reuses" `CaptureInitializationResult` as the return type. Could it just return `grpc::Status`? I guess the thing was that we wanted to keep `LinuxCaptureServiceBase` "grpc-free"? But maybe it's fine as it's just a status. Otherwise we might need a new "result" type for `DoCapture`/

Reply:
How about I move both `InitializeCapture` and `WaitForStart` from `LinuxCaptureServiceBase::DoCapture`,
and change both Windows/Linux capture services' `Capture` rpc to have the following workflow
https://screenshot.googleplex.com/C6ifwiNe3pTePyN

Reply:
> Maybe the problem comes from the fact that DoCapture "reuses" `CaptureInitializationResult` as the return type. Could it just return `grpc::Status`?

This also sounds good to me. Thanks for the suggestion!